File: LittleProjects/Java/SpringBootCassandraCrudWithRedisCache/src/main/java/com/github/gr1f0n6x/springbootcassandracrud/service/Service.java
package com.github.gr1f0n6x.springbootcassandracrud.service;

import java.util.List;

public interface Service<E, PK> {
    E findById(PK pk);
    List<E> findAll();
    void deleteById(PK pk);
    void deleteAll();
    E save(E e);
    E update(E e);
}

File: LittleProjects/Scala/ScalatraDoobieCrud/scalatradoobiecrud/src/main/scala/com/github/gr1f0n6x/scalatra/doobie/crud/service/Service.scala
package com.github.gr1f0n6x.scalatra.doobie.crud.service

trait Service[A, ID] {
  def findOne(id: ID): Either[Throwable, A]

  def findAll(): Either[Throwable, Seq[A]]

  def save(a: A): Either[Throwable, Any]

  def update(a: A): Either[Throwable, Any]

  def deleteOne(id: ID): Either[Throwable, Any]

  def deleteAll(): Either[Throwable, Any]
}
